interactions/models/discord/guild.py

```diff
@@ -1457,7 +1457,7 @@ async def create_role(
         if name:
             payload["name"] = name
 
-        if permissions:
+        if permissions or permissions == Permissions.NONE:
```

Comment:
It's probably better to do `permissions is not MISSING` for the entire statement instead, instead of awkwardly working around a truthy.

Reply:
Cool, yeah that sense. I'll make that change.

Reply:
Actually I just noticed the type hint on this function indicates `Optional` as well. Do you think `permissions is not MISSING and permissions is not None` would be more appropriate?

Reply:
The optionals seem like a mistake looking at it, but that's a tad bit out of the scope of the PR. I'd say go for the change.

Reply:
Great I've made the change. And thanks for the quick review 🙏